### PR TITLE
Add Support To Edit An Existing Plan

### LINKED ISSUE #452: Add Support To Edit An Existing Plan

### PR DESCRIPTION
**Pull request by rael346**
# Description

Add an Edit Plan button that edit the name, catalog year and the major of the current plan

https://user-images.githubusercontent.com/75349763/196054929-72af0aff-65a4-48e0-aefb-5a6da41d9da4.mov

Closes #452

## Type of change
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update
- [ ] This requires a run of `yarn install`

# How Has This Been Tested?
- [x] Adding a new plan using the new plan modal and edit it using the edit plan modal 
- [ x ] Checking the data changes using Postman

# Checklist:

- [x] I have performed a self-review of my own code
- [x] I have commented my code where needed 
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: packages/frontend-v2/components/Plan/EditPlanModal.tsx [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+  });
+
+  useEffect(() => {
+    // Change the default field to the corresponding plan when a plan is selected/edited
+    reset({
+      name: plan?.name,
+      catalogYear: plan?.catalogYear,
+      major: plan?.major,
+    });
+  }, [plan, reset]);
```
Comment (aryanshah701) [inline packages/frontend-v2/components/Plan/EditPlanModal.tsx:71]: nice!!

File: packages/frontend-v2/components/Plan/EditPlanModal.tsx [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+
+    try {
+      await API.plans.update(planId, payload);
+    } catch (error) {
+      handleApiClientError(error as Error, router);
+      return;
+    }
+
+    mutate(USE_STUDENT_WITH_PLANS_SWR_KEY);
+    toast.success("Update Plan Success");
```
Comment (aryanshah701) [inline packages/frontend-v2/components/Plan/EditPlanModal.tsx:87]: <3

File: packages/frontend-v2/pages/home.tsx [hunk 3]
```diff
@@ -175,27 +168,6 @@ const HomePage: NextPage = () => {
...
-}) => {
-  return (
-    <>
-      <BlueButton leftIcon={<AddIcon />} onClick={onOpen} ml="xs" size="md">
-        Add Plan
-      </BlueButton>
-      <AddPlanModal onClose={onClose} isOpen={isOpen} />
-    </>
-  );
-};
```
Comment (aryanshah701) [inline packages/frontend-v2/pages/home.tsx:197]: thank you!

File: packages/frontend-v2/pages/home.tsx [hunk 4]
```diff
@@ -140,12 +134,12 @@ const HomePage: NextPage = () => {
...
-      setSelectedPlanId(newPlanId);
-    }
-    closeAddPlanModalDisplay();
-  };
+  // const onCloseAddPlanModal = (newPlanId?: number) => {
+  //   if (newPlanId) {
+  //     setSelectedPlanId(newPlanId);
+  //   }
+  //   closeAddPlanModalDisplay();
+  // };
```
Comment (aryanshah701) [inline packages/frontend-v2/pages/home.tsx:143]: can remove this :)

**Review by aryanshah701 — APPROVED** [review]
Awesome!! 🙌 Love the new Form components <3

Gotta resolve some merge conflicts but good to ship 🚢

File: packages/frontend-v2/pages/home.tsx [hunk 5]
```diff
@@ -141,25 +135,6 @@ const HomePage: NextPage = () => {
...
   };
 
-  const onCloseAddPlanModal = (newPlanId?: number) => {
-    if (newPlanId) {
-      setSelectedPlanId(newPlanId);
-    }
-    closeAddPlanModalDisplay();
-  };
-
-  const onCloseDeletePlanModal = (
```
Comment (rael346) [inline packages/frontend-v2/pages/home.tsx:151]: Refactored this logic to `DeletePlanModal` component

**Pull request merged**